psl-core/src/main/java/org/linqs/psl/database/Database.java

```diff
@@ -76,191 +88,42 @@
-	 *	ObservedAtom if its Predicate is closed and as a {@link RandomVariableAtom}
-	 *	if it is open.</li>
-	 *	<li>If the GroundAtom has a StandardPredicate but is not persisted
-	 *	in any of the Database's partitions, it will be instantiated with a truth
-	 *	value of 0.0. It will be instantiated as an ObservedAtom if its Predicate
-	 *	is closed and as a RandomVariableAtom
-	 *	if it is open.</li>
-	 *	<li>If the GroundAtom has an ExternalFunctionalPredicate, then it will be
-	 *	instantiated as an ObservedAtom with the functionally defined
-	 *	truth value.</li>
-	 * </ul>
-	 *
-	 * @param predicate the Predicate of the Atom
-	 * @param arguments the GroundTerms of the Atom
-	 * @return the Atom
-	 * @throws IllegalArgumentException if predicate is not registered or arguments are not valid
-	 * @throws IllegalStateException if the Atom is persisted in multiple read Partitions
-	 */
-	public GroundAtom getAtom(Predicate predicate, Constant... arguments);
+	public abstract GroundAtom getAtom(Predicate predicate, Constant... arguments);
 
-	/**
-	 * Check to see if a ground atom exists in the database.
-	 * This looks for a real ground atom and ignores the closed-world assumption.
-	 * If found, the atom will be cached for subsequent requests to this or getAtom().
-	 */
-	public boolean hasAtom(StandardPredicate predicate, Constant... arguments);
+	public abstract boolean hasAtom(StandardPredicate predicate, Constant... arguments);
 
-	/**
-	 * Get a count of all the ground atoms for a predicate.
-	 * By "ground", we mean that it exists in the database.
-	 * This will not leverage the closed world assumption for any atoms.
-	 *
-	 * @param predicate the predicate to get a count for
-	 * @return The count of all ground atoms present in any partition of the database.
-	 */
-	public int countAllGroundAtoms(StandardPredicate predicate);
+	public abstract int countAllGroundAtoms(StandardPredicate predicate);
 
-	/**
-	 * Get a count of all the ground RandomVariableAtoms for a predicate.
-	 * By "ground", we mean that it exists in the database.
-	 * This will not leverage the closed world assumption for any atoms.
-	 *
-	 * @param predicate the predicate to get a count for
-	 * @return The count of all ground atoms present in the write partition of the database.
-	 */
-	public int countAllGroundRandomVariableAtoms(StandardPredicate predicate);
+	public abstract int countAllGroundRandomVariableAtoms(StandardPredicate predicate);
 
-	/**
-	 * Fetch all the ground atoms for a predicate.
-	 * By "ground", we mean that it exists in the database.
-	 * This will not leverage the closed world assumption for any atoms.
-	 *
-	 * @param predicate the predicate to fetch atoms for
-	 * @return All ground atoms present in any partition of the database.
-	 */
-	public List<GroundAtom> getAllGroundAtoms(StandardPredicate predicate);
+	public abstract List<GroundAtom> getAllGroundAtoms(StandardPredicate predicate);
```

Comment:
Make getAllGroundAtoms(StandardPredicate predicate, List<Integer> partitions) an abstract method here.
Then you can move this implementation up to here (Database.java).